### PR TITLE
kvserver: unskip `TestNewTruncateDecision`

### LINKED ISSUE #38584: storage: TestNewTruncateDecision failed under stress [skipped]
SHA: https://github.com/cockroachdb/cockroach/commits/ca1ef4d4f8296b213c0b2b140f16e4a97931e6e7

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNewTruncateDecision PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1368520&tab=buildLog

```
=== RUN   TestNewTruncateDecision
I190701 12:12:45.945806 12152 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190701 12:12:45.946055 12152 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
--- FAIL: TestNewTruncateDecision (0.24s)
    raft_log_queue_test.go:570: truncation should not have occurred, but truncatableIndexes changed from 1 -> 2
    raft_log_queue_test.go:573: truncation should not have occurred, but oldestIndex changed from 116 -> 117

```

Jira issue: CRDB-5625

### PR DESCRIPTION
**Pull request by erikgrinaker**
Passed after 10k stress runs. Has been skipped since 2019, issue seems to have been fixed in the meanwhile.

Resolves #38584.

Epic: none
Release note: None

### CODE REVIEW

**Review by pav-kv — APPROVED** [review]

**Comment by erikgrinaker** [comment]
Saw this fail under stressrace:

```
    raft_log_queue_test.go:557: truncation should not have occurred, but truncatableIndexes changed from 1 -> 2
    raft_log_queue_test.go:560: truncation should not have occurred, but oldestIndex changed from 116 -> 117
```

**Comment by erikgrinaker** [comment]
Disabling quiescence, as indicated in https://github.com/cockroachdb/cockroach/issues/38584#issuecomment-508082067, fixed it under stressrace too. Not pretty, but it works.

File: pkg/kv/kvserver/raft_log_queue_test.go [hunk 1]
```diff
@@ -450,7 +449,11 @@ func TestNewTruncateDecision(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 38584)
+	// Unquiescing can add spurious empty log entries. Just disable it.
+	testingDisableQuiescence = true
```
[the remark this answers is not in the document]
Comment (erikgrinaker) [reply pkg/kv/kvserver/raft_log_queue_test.go:453]: We don't run tests in parallel in the same package, on separate packages (which are run as separate binaries).

[the remark this answers is not in the document]
Comment (pavelkalinnikov) [reply pkg/kv/kvserver/raft_log_queue_test.go:453]: Does `dev test --stress` run separate binaries too?

**Review by pav-kv — APPROVED** [review]
Still LGTM, assuming parallel test runs aren't possible.

File: pkg/kv/kvserver/raft_log_queue_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (erikgrinaker) [reply pkg/kv/kvserver/raft_log_queue_test.go:453]: Yes.

[the remark this answers is not in the document]
Comment (erikgrinaker) [reply pkg/kv/kvserver/raft_log_queue_test.go:453]: FWIW, to run Go tests in parallel, they must explicitly opt in to it by calling `t.Parallel()`. We only do this in a handful of packages, and not in `kvserver`.

https://pkg.go.dev/testing#T.Parallel

**Pull request merged**